File: LeetCode/1047. Remove All Adjacent Duplicates In String.py
class Solution:
    def removeDuplicates(self, s: str) -> str:
        stack = []

        for i in s:
            if not stack: #if stack is empty
                stack.append(i)
            else:
                if(i==stack[-1]):
                    stack.pop()
                else:
                    stack.append(i)

        return ''.join(stack)

File: LeetCode/1047. Remove All Adjacent Duplicates In String.cpp
class Solution {
public:
    string removeDuplicates(string s) {
        string stack;
        
        for (char c : s) {
            if (!stack.empty() && c == stack.back()) {
                stack.pop_back();
            } 
            else {
                stack.push_back(c);
            }
        }
        
        return stack;
        
    }
};
